File: app/src/main/java/org/test/kotlin/demo/repos/search/SearchReposContract.java
package org.test.kotlin.demo.repos.search;

import org.test.kotlin.demo.api.dto.RepositoryDTO;

import java.util.List;

import androidx.annotation.NonNull;

interface SearchReposContract {

    interface View {

        void showRepositories(@NonNull List<RepositoryDTO> repositories);

        void showError(Throwable throwable);

    }

    interface Presenter {

        void onQueryChange(@NonNull String query);

    }

}

File: app/src/main/java/org/test/kotlin/demo/repos/search/SearchReposContract.kt
package org.test.kotlin.demo.repos.search

import org.test.kotlin.demo.api.dto.RepositoryDTO

interface SearchReposContract {

    interface View {

        fun showRepositories(repositories: List<RepositoryDTO>)

        fun showError(throwable: Throwable)

    }

    interface Presenter {

        fun onQueryChange(query: String)

    }

}
